### ISSUE TITLE
Lost and confused with getting addons working.

### ISSUE THREAD

**Issue by brycedarling** [jekyll/jekyll-assets#221]
The project I'm having an issue with can be found here https://github.com/brycedarling/foo

After some trial and error I figured out that I need to have gem "uglifier" in my Gemfile in order to get jekyll-assets to compress my JS files. However, I can't seem to get it to play nicely with sprockets-es6. I get the following error:

jekyll 3.0.1 | Error:  Unexpected token: punc ()) (line: 1, col: 12, pos: 12)

Which appears to be happening because sprockets-es6 isn't running before uglifier. It doesn't appear to be running at all actually, even if I comment out the uglifier gem the ES6 code isn't being transpiled.

I'm also struggling to get autoprefixer and mini_magick to work. No errors, but no prefixes being added to my CSS or SASS and no image compression happening either. When I don't include autoprefixer-rails and sprockets-es6 in the Gemfile I do see the warning message about them not being able to be required.

It's probably some user error but after being stuck with getting this to work for awhile and not making any progress by re-re-reading docs or source I figured I'd just ask for help. Thanks!

**Comment by envygeeks** [comment]
That source does work for me, as it stands, except here are a few notes so we can head towards diagnostic relevancy:

* Do you have NodeJS installed? Or if not: therubyracer? We do not support Ruby Rhino because Rhino JavaScript engine is broken with many of these addons which seem to prefer JavaScriptCore from Webkit.  We used to carry a warning here and on Jekyll Docker but no longer bother.

* Did you compare the image sizes for the mini_magic? The image moves from 21mb to 13mb for me, that's still hella hefty but it did process it for me.  The mini_magic gem requires you to install imagemagic on the system.  If it didn't throw during that process that's a mini_magic bug but I'd be happy to look into solving it here but it *should* throw somewhere in the pipeline if it can't find imagemagic of some kind.

* Where are you trying to prefix? I noticed `foo.scss` had what most would consider the need to prefix (border-radius) but that is supported in IE10, Edge, Chrome, Firefox, Opera and most every other brower based on one of the engines that powers the former so it's not prefixed.  I do get a prefix when I add "order: 1" to the foo.scss I do see it prefix that because I believe only Chrome and Edge have brought that ouf of their unstable usage.
  * Right now we let autoprefixer ride it's defaults because there were some unacceptable options that required me to build a configuration processor like Jekyll (such as people's abilities to use Filesystem paths outside of our path.) I believe the defaults are setup for the *modern web* not the old hindering web.

* As for ES6, I do not see any ES6 in that source at all.  I do see the message to Hannah (too bad it's not to [Hanna][Hanna], maybe should could come and teach the system a lesson.)  Now, when I edit the source and add the source in [es6-example](#es6-example) and set the file extension to "js.es6" as required by Sprockets and ES6 I end up with the pollyfills and converted source.

#### ES6 Example

```js
setTimeout(() => {
  console.log('hello, hannah');
}, 1000);

class Polygon {
  constructor(height, width) {
    this.height = height;
    this.width = width;
  }

  get area() {
    return this.calcArea()
  }
}
```

[Hanna]: https://en.wikipedia.org/wiki/Hanna_(film)

**Comment by envygeeks** [comment]
TLDR of the above:

* Do you have `nodejs` (on the system) or `therubyracer` (in Gemfile) installed?
* Do you have `imagemagick` installed on the system? Compare the image sizes and data to confirm.
* Autoprefixer sanely defaults to modern web prefixing. Try something like `order` to test.
* ES6 processor only processes the ".js.es6" or ".es6" extension, ensure there is ES6.

**Comment by brycedarling** [comment]
Good news, autoprefixer is working when I added the order property AND sprockets-es6 is working once I renamed the file extension to .js.es6. Thank you!

The ES6 code I had in there was only the fat arrow syntax but it got transpiled successfully. With autoprefixer, I guess I was expecting any/all css properties that could have prefixes to get them added (ie -webkit-font-smoothing and the -moz-border-radius).

Yeah, the image file is massive and uncompressed on purpose so I could see the image file drop in size. I just tried installing imagemagick via "brew install imagemagick" but the image still stays at 21mb for me. Also, is there a way to set a max image width and have it automatically keep the aspect ratio?

**Comment by brycedarling** [comment]
It does seem as though mini_magick is working in an irb session:

$ irb
irb(main):001:0> require 'mini_magick'
=> true
irb(main):002:0> image = MiniMagick::Image.new("header2.png")
=> #<MiniMagick::Image:0x007fca6087a9b0 @path="header2.png", @tempfile=nil, @info=#<MiniMagick::Image::Info:0x007fca6087a988 @path="header2.png", @info={}>>
irb(main):003:0> puts image[:width]
4000

**Comment by envygeeks** [comment]
Try clearing the cache @brycedarling, delete .asset-cache, we do cache anything processed through any of our addons and if the hash hasn't changed then we don't reprocess, we do this so you don't wait 10 hours each process.  There really is no better way to determine the caching imo, well we could also detect if the image changed but that's probably not what's happening here.

On that note, I do not know much about imagemagic but the good news here is that any option I provide for image magic, is a straight up pass-through, I don't validate it, I don't care to validate it, I let magic crash for you, so anything that I pass through is literally what imagemagic itself would support so (and I could be wrong on this value, you'll have to look it up in the imagemagic docs) `{% img name resize:'80x80>' %}` should work because we just pass through '80x80>' and let Minimagick crash.

**Comment by brycedarling** [comment]
Fantastic! That was exactly what it needed to get it working. My file only compressed to 17.7mb and not 13 like yours, but I'll take it! Once I cap the image width's everything will be fine there anyways.

Thanks man, your help is really appreciated. This plugin makes Jekyll awesome. :+1:

**Comment by envygeeks** [comment]
Glad you got it working and thanks.  Closing this ticket as everything is resolved.
